src/devices/device_models/thermal_generation.jl

```diff
@@ -464,15 +520,37 @@ function time_constraints!(canonical_model::CanonicalModel,
 
     parameters = model_has_parameters(canonical_model)
     resolution = model_resolution(canonical_model)
-    duration_data = _get_data_for_tdc(devices, resolution)
+    
+    key_on =  Symbol("duration_on_$(T)")
+    key_off =  Symbol("duration_off_$(T)")
 
-    if !(isempty(duration_data[1]))
-        key_on =  Symbol("duration_on_$(T)")
-        key_off =  Symbol("duration_off_$(T)")
-        if !(key_on in keys(canonical_model.initial_conditions))
-            @warn("Initial Conditions for Time Up/Down constraints not provided. This can lead to unwanted results")
-            time_limits = duration_init(canonical_model, devices, duration_data[1])
+    if !(key_on in keys(canonical_model.initial_conditions))
+        @warn("Initial Conditions for Minimum Up Time constraints not provided. This can lead to unwanted results")
+        duration_init_on(canonical_model, devices)
+    else
+        dur_on_miss = missing_init_cond(canonical_model.initial_conditions[key_on], devices)
+        if !isnothing(dur_on_miss)
+            @warn("Initial Conditions for Minimum Up Time constraints not provided. This can lead to unwanted results")
+            duration_init_on(canonical_model, dur_on_miss)
         end
+    end
+
+    if !(key_off in keys(canonical_model.initial_conditions))
+        @warn("Initial Conditions for Minimum Down Time constraints not provided. This can lead to unwanted results")
+        duration_init_off(canonical_model, devices)
+    else
+        dur_off_miss = missing_init_cond(canonical_model.initial_conditions[key_off], devices)
+        if !isnothing(dur_off_miss)
+            @warn("Initial Conditions for Minimum Down Time constraints not provided. This can lead to unwanted results")
+            duration_init_off(canonical_model, dur_off_miss)
+        end
+    end
+
+    sort!(canonical_model.initial_conditions[key_on],by=x-> PSY.get_name(x.device))
```

Comment:
can we avoid this sort?

Reply:
That the only way I found to make sure both the initial conditions arrays have the same ordering. I told Dan this problem and this is what he recommended. If there is a better way to do it, then I'll definitely change this